### ISSUE TITLE
Switching over to using `@testset`s

### ISSUE THREAD

**Issue by Rory-Finnegan** [JuliaStats/StatsBase.jl#251]
The `@testset` macro (introduced in 0.5) provides nicer test reporting.

**Comment by ararslan** [comment]
Yes, this would be good. It's been on my to-do list for a while but I haven't gotten around to it. A PR would be welcome if you're interested in taking it on.

**Comment by Rory-Finnegan** [comment]
Sure. I'm going to need to change a bunch of tests for #250, so maybe I'll just roll it in with that.

**Comment by ararslan** [comment]
That'd be fine, though having it as a separate PR (or at least a separate commit) would make it easier to review.

**Comment by Rory-Finnegan** [comment]
Yeah, I was planning on just adding an extra commit once that PR is ready.

**Comment by sbromberger** [comment]
Just a heads-up - if you wrap all your tests in nested testsets, you won't see progress in Travis - the output is delayed until all tests are complete. (This can be a disadvantage.)
